### PR TITLE
re-enable PTH rules for ruff

### PR DESCRIPTION
**Pull request by dgarros**
This PR fixes the following rules that were previously ignored by ruff
- "PTH100",  # `os.path.abspath()` should be replaced by `Path.resolve()`
- "PTH102",  # `os.mkdir()` should be replaced by `Path.mkdir()`
- "PTH107",  # `os.remove()` should be replaced by `Path.unlink()`
- "PTH108",  # `os.unlink()` should be replaced by `Path.unlink()`
- "PTH109",  # `os.getcwd()` should be replaced by `Path.cwd()`
- "PTH110",  # `os.path.exists()` should be replaced by `Path.exists()`
- "PTH112",  # `os.path.isdir()` should be replaced by `Path.is_dir()`
- "PTH113",  # `os.path.isfile()` should be replaced by `Path.is_file()`
- "PTH117",  # `os.path.isabs()` should be replaced by `Path.is_absolute()`
- "PTH118",  # `os.path.join()` should be replaced by `Path` with `/` operator

### CODE REVIEW

File: backend/infrahub/git/integrator.py [hunk 1]
```diff
@@ -414,7 +413,7 @@ async def get_repository_config(self, branch_name: str, commit: str) -> Optional
         branch_wt = self.get_worktree(identifier=commit or branch_name)
 
         config_file_name = ".infrahub.yml"
-        config_file = Path(os.path.join(branch_wt.directory, config_file_name))
+        config_file = Path(branch_wt.directory, config_file_name)
```
Comment (gmazoyer) [inline backend/infrahub/git/integrator.py]: `branch_wt.directory` should already be a `Path` object so `config_file = branch_wt.directory / config_file_name` should be enough.

File: backend/tests/integration/conftest.py [hunk 2]
```diff
@@ -50,9 +50,9 @@ async def load_infrastructure_schema(db: InfrahubDatabase):
     tmp_schema = branch_schema.duplicate()
 
     for file_name in os.listdir(base_dir):
-        file_path = os.path.join(base_dir, file_name)
+        file_path = Path(base_dir, file_name)
 
-        if file_path.endswith((".yml", ".yaml")):
+        if str(file_path).endswith((".yml", ".yaml")):
```
Comment (gmazoyer) [inline backend/tests/integration/conftest.py]: `file_path.suffix in (".yml", ".yaml")`

File: backend/tests/integration/git/test_git_repository.py [hunk 3]
```diff
@@ -35,9 +35,9 @@ async def load_infrastructure_schema(db: InfrahubDatabase):
     tmp_schema = branch_schema.duplicate()
 
     for file_name in os.listdir(base_dir):
-        file_path = os.path.join(base_dir, file_name)
+        file_path = Path(base_dir, file_name)
 
-        if file_path.endswith((".yml", ".yaml")):
+        if str(file_path).endswith((".yml", ".yaml")):
```
Comment (gmazoyer) [inline backend/tests/integration/git/test_git_repository.py]: `file_path.suffix in (".yml", ".yaml")`

File: backend/tests/unit/git/conftest.py [hunk 4]
```diff
@@ -660,9 +659,7 @@ async def gql_query_data_02() -> dict:
 
 @pytest.fixture
 async def mock_schema_query_01(helper, httpx_mock: HTTPXMock) -> HTTPXMock:
-    response_text = Path(os.path.join(helper.get_fixtures_dir(), "schemas", "schema_01.json")).read_text(
-        encoding="UTF-8"
-    )
+    response_text = Path(helper.get_fixtures_dir(), "schemas", "schema_01.json").read_text(encoding="UTF-8")
```
Comment (gmazoyer) [inline backend/tests/unit/git/conftest.py]: Again `helper.get_fixtures_dir()` is a `Path` object, weird that it was used with `os.path.join`

File: backend/tests/unit/git/test_git_repository.py [hunk 5]
```diff
@@ -124,10 +123,10 @@ async def test_init_existing_repository(git_repo_01: InfrahubRepository):
...
     # Check if all the directories are present
     assert repo.has_origin is True
-    assert os.path.isdir(repo.directory_root)
-    assert os.path.isdir(repo.directory_branches)
-    assert os.path.isdir(repo.directory_commits)
-    assert os.path.isdir(repo.directory_temp)
+    assert Path.is_dir(repo.directory_root)
+    assert Path.is_dir(repo.directory_branches)
+    assert Path.is_dir(repo.directory_commits)
+    assert Path.is_dir(repo.directory_temp)
```
Comment (gmazoyer) [inline backend/tests/unit/git/test_git_repository.py]: `repo.directory_*` are `Path` objects, so `repo.directory_root.is_dir()` should be fine

File: backend/tests/integration/conftest.py (hunk 2 above)
Comment (dgarros) [reply backend/tests/integration/conftest.py]: Thank you
I knew there was a better way but I was on the plane without internet connection while I did this PR :/

**Review by ajtmccarty — APPROVED** [review]

**Review by gmazoyer — APPROVED** [review]

**Pull request merged**